src/Workspaces/CSharp/Portable/Extensions/ContextQuery/CSharpSyntaxContext.cs

```diff
@@ -340,5 +340,15 @@ internal override ITypeInferenceService GetTypeInferenceServiceWithoutWorkspace(
         {
             return new CSharpTypeInferenceService();
         }
+
+        /// <summary>
+        /// Is this a possible position for an await statement (`await using` or `await foreach`)?
+        /// </summary>
+        internal bool IsAwaitStatementContext(int position, CancellationToken cancellationToken)
+        {
+            var leftToken = this.SyntaxTree.FindTokenOnLeftOfPosition(position, cancellationToken);
+            var targetToken = leftToken.GetPreviousTokenIfTouchingWord(position);
+            return targetToken.Kind() == SyntaxKind.AwaitKeyword && targetToken.Parent.IsParentKind(SyntaxKind.ExpressionStatement);
```

Comment:
Done. Not sure which is better though (seems equivalent).

Reply:
Difference being that one depends on the exact shape of the tree (which may change depending on things like error tolerance and whatnot), and the other is abstraction that more conveys intent.  i.e. "i can write `using/foreach` after an 'await' if the previous token is an 'await' and that await started a statement".  the "started a statement" part is the interesting thing since there's really no guarantee that that will definitely be a an expression-statement afaict.

Reply:
Makes sense. Thanks for the explanation.